### ISSUE TITLE
TodoByVersionRule should not match date comments

### ISSUE THREAD

**Issue by staabm** [staabm/phpstan-todo-by#59]
the pattern in `TodoByVersionRule` is not optimal, which means the rule will trigger on the following example

```
<?php
//  TODO: 2022-12-08 Foo
```

only the `TodoByDateRule` should be triggered in this case.
